pydantic/main.py

```diff
@@ -1158,14 +1160,16 @@ def parse_file(  # noqa: D102
         )
         from .deprecated import parse
 
-        obj = parse.load_file(
-            path,
-            proto=proto,
-            content_type=content_type,
-            encoding=encoding,
-            allow_pickle=allow_pickle,
-        )
-        return cls.parse_obj(obj)
+        with warnings.catch_warnings():
```

Comment:
I definitely don't think we want to add these in `pydantic/main.py`. What warnings are being raised?

Reply:
A warning about `load_file()` (called below) being deprecated, possibly more.

Reply:
I don't think there's a point in rewriting a method that's deprecated itself, at this point.

Reply:
Well I mean we should use whatever the new api is...

Reply:
Oh. Do you mean we should remove tests for deprecated API then?

Reply:
Yeah, we should remove the catching of the warnings and write better code so that we don't have to catch warnings. I haven't done much research into what new calls we should make instead, though.